fetch_tags.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+###
+# EXIT CODES
+# 0 - success
+# 1 - parameter error
+# 50 - dockerhub pull error
+# 51 - no valid docker image tag
+# 52 - no old tag file found
+# 53 - max tag count iteration reached
+# 100 - new tag found, pr_script triggered
+###
+
+# Sample envionment variables
+# IMAGE_ID="weaver-worker"
+# DOCKERHUB_REPO="pavics/weaver"
+# TAG_FILTER="^[0-9]+(\\.[0-9]+)*$"
+# BUMP_TAG="WEAVER_WORKER_IMAGE"
+# BUMP_FILE="birdhouse/default.env"
+
+REQUIRED_ENV_VARS='
+    IMAGE_ID
+    DOCKERHUB_REPO
+    TAG_FILTER
+    BUMP_TAG
+    BUMP_FILE
+'
+
+# args parsing
+for env_var in $REQUIRED_ENV_VARS
+do
+    if [[ ! -v "${env_var}" ]]; then
+        echo "[ERROR] [$0] Missing ${env_var} environment variable. Exiting."
+        exit 1
+    fi
+done
+
+echo "[STEP] [$0] [$IMAGE_ID] Fetch tags"
+
+DATA_DIR="data"
+
+
+# get latest image tag from dockerhub
+mkdir -p data
+NAME=${DOCKERHUB_REPO//\//_}_$IMAGE_ID
+NEW_FILENAME=$NAME.new
+NEW_FILEPATH=$DATA_DIR/$NEW_FILENAME
+DOCKERHUB_IMAGE_TAGS=$(wget -q $DOCKERHUB_IMAGE_TAGS_URL$DOCKERHUB_REPO/tags?page_size=1024 -O-)
+MAX_TAG_ITERATION=$(echo $DOCKERHUB_IMAGE_TAGS | jq '.count')
+TAG_INDEX=0
+TAG_FILTER=${TAG_FILTER//\\\\/\\}    # replace \\ with \ in regex
+
+while [[ ! $LATEST_TAG =~ $TAG_FILTER ]]
```

Comment:
I would like to see a small comment describing code blocks like "Get latest tag from dockerhub". 
Came back to add that however most of the code blocks already have comments, thanks!